core/core-impl/src/main/java/ai/timefold/solver/core/api/solver/SolverJobBuilder.java

```diff
@@ -0,0 +1,95 @@
+package ai.timefold.solver.core.api.solver;
+
+import java.util.UUID;
+import java.util.function.BiConsumer;
+import java.util.function.Consumer;
+import java.util.function.Function;
+
+import ai.timefold.solver.core.api.domain.solution.PlanningSolution;
+
+/**
+ * Provides a fluid contract that allows customization and submission of planning problems to solve.
+ * <p>
+ * A {@link SolverManager} can solve multiple planning problems and can be used across different threads. The builder is
+ * responsible for constructing and solving a specific configuration problem. Hence, it is possible to
+ * have multiple distinct build configurations that are scheduled to run by the {@link SolverManager} implementation.
+ * <p>
+ * To solve a planning problem, set the problem configuration: {@link #withProblemId(Object)},
+ * {@link #withProblemFinder(Function)} and {@link #withProblem(Object)}. Then solve it by calling {@link #run()}.
```

Comment:
Typically we put each sentence on a new line, to make the comment read more easily. If you install the Grazie plugin in your IDE, it will suggest where to put line breaks. (Among other things.)